apps/eth_blockchain/lib/eth_blockchain/adapter_server.ex

```diff
@@ -0,0 +1,202 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EthBlockchain.AdapterServer do
+  @moduledoc false
+
+  @typep server :: GenServer.server()
+  @typep from :: GenServer.from()
+  @typep state :: {atom(), map()}
+
+  @typep adapter :: EthBlockchain.adapter()
+  @typep call :: EthBlockchain.call()
+  @typep mfargs :: {module(), atom(), [term()]}
+
+  @typep resp(ret) :: ret | {:error, atom()}
+  @typep reply(ret) :: {:reply, resp(ret), state()}
+
+  use GenServer
+  require Logger
+
+  @doc """
+  Starts EthBlockchain.Adapter.
```

Comment:
```suggestion
  Starts EthBlockchain.AdapterServer.
```

also the rest of the module